h/streamer/messages.py

```diff
@@ -74,10 +74,23 @@ def handle_message(message, settings, session, topic_handlers):
     # to stop connections being added or dropped during iteration, and if that
     # happens Python will throw a "Set changed size during iteration" error.
     sockets = list(websocket.WebSocket.instances)
-    handler(message.payload, sockets, settings, session)
+
+    # The `prepare` function sets the active registry which is an implicit
+    # dependency of some of the authorization logic used to look up annotation
+    # and group permissions.
+    with pyramid.scripting.prepare(registry=registry):
+        handler(message.payload, sockets, registry, session)
+
+
+def handle_user_event(message, sockets, registry, session):
+    for socket in sockets:
+        reply = _generate_user_event(message, socket)
+        if reply is None:
+            continue
+        socket.send_json(reply)
```

Comment:
Moved this up so that `handle_annotation_event` and it's private function `_generate_annotation_event` are next to each other.